media-gfx/tuxpaint/tuxpaint-0.9.24.ebuild

```diff
@@ -0,0 +1,78 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs xdg-utils desktop
+
+DESCRIPTION="Drawing program designed for young children"
+HOMEPAGE="http://www.tuxpaint.org/"
+SRC_URI="mirror://sourceforge/${PN}/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE=""
+
+RDEPEND="
+	app-text/libpaper
+	dev-libs/fribidi
+	gnome-base/librsvg:2
+	>=media-libs/libpng-1.2:0=
+	>=media-libs/freetype-2:2
+	media-libs/libsdl
+	media-libs/sdl-image[png]
+	media-libs/sdl-mixer
+	media-libs/sdl-pango
+	media-libs/sdl-ttf
+	x11-libs/cairo
+	"
+DEPEND="${RDEPEND}"
+BDEPEND="sys-devel/gettext
+	media-gfx/imagemagick[png]
+	dev-util/gperf
+	"
+
+src_prepare() {
+	default
+	sed -i \
+		-e 's|linux_ARCH_INSTALL:=install-xdg|linux_ARCH_INSTALL:=|' \
+		-e "s|linux_PREFIX:=/usr/local|linux_PREFIX:=/usr|" \
+		-e "s:/lib/:/$(get_libdir)/:g" \
+		-e 's:/share/doc/tuxpaint-$(VER_VERSION)/:'"/share/doc/${PF}/:g" \
+		-e '/@gzip -f/d' \
+		-e '/@chmod a+rx,g-w,o-w $(MAN_PREFIX)/d' \
+		-e "s|linux_ARCH_CFLAGS:=|linux_ARCH_CFLAGS:= ${CFLAGS}|" \
+		Makefile || die
+}
+
+src_compile() {
+	# parallel build may break things
+	emake -j1 CC="$(tc-getCC)"
+}
+
+src_install() {
+	emake -j1 DESTDIR="${D}" install
+	local file size
+	for file in data/images/icon[0-9]*x[0-9]*.png; do
+		size=${file##*/icon}
+		size=${size%%x*}
+		newicon -s "${size}" "${file}" tux4kids-tuxpaint.png
+	done
+	newmenu src/tuxpaint.desktop tux4kids-tuxpaint.desktop
+	dodoc docs/*.txt
+	dodoc docs/en/*.txt
+}
+
+pkg_postinst() {
+	xdg_icon_cache_update
+
+	if ! has_version "${CATEGORY}/${PN}"; then
+		elog "For additional graphic stamps, you can emerge the"
```

Comment:
You might want [optfeature.eclass](https://devmanual.gentoo.org/eclass-reference/optfeature.eclass/index.html)!

Reply:
optfeature is a bit more suited to something that has more than one optional dependency, I think. I will tweak the logic to work like it though.